flac.go

```diff
@@ -327,60 +314,44 @@ func (stream *Stream) Seek(sampleNum int64, whence int) (result int64, err error
 
 	rs := stream.r.(io.ReadSeeker)
 
-	var point meta.SeekPoint
-	switch whence {
-	case io.SeekStart:
-		point = stream.searchFromStart(sampleNum)
-	case io.SeekCurrent:
-		point, err = stream.searchFromCurrent(sampleNum, rs)
-	case io.SeekEnd:
-		point = stream.searchFromEnd(sampleNum)
-	default:
-		return 0, ErrInvalidSeek
+	isBiggerThanStream := stream.Info.NSamples != 0 && stream.Info.NSamples < sampleNum
```

Comment:
Personal preference, since we named the variable isBiggerThanStream, then `sampleNum > NSamples` reads better in my mind.

```suggestion
	isBiggerThanStream := stream.Info.NSamples != 0 && sampleNum > stream.Info.NSamples
```